maas/resource_maas_boot_source.go

```diff
@@ -1,12 +1,146 @@
 package maas
 
 import (
+	"context"
 	"fmt"
 
 	"github.com/canonical/gomaasclient/client"
 	"github.com/canonical/gomaasclient/entity"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
+func resourceMAASBootSource() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a resource to manage the MAAS boot source.",
+		CreateContext: resourceBootSourceCreate,
+		ReadContext:   resourceBootSourceRead,
+		UpdateContext: resourceBootSourceUpdate,
+		DeleteContext: resourceBootSourceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"created": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The creation time of the boot source.",
+			},
+			"keyring_data": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "The data on the keyring for the boot source.",
+				ConflictsWith: []string{"keyring_filename"},
+			},
+			"keyring_filename": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "The filename on the keyring for the boot source.",
+				ConflictsWith: []string{"keyring_data"},
+			},
+			"updated": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The time of most recent update of the boot source.",
+			},
+			"url": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The URL of the boot source.",
+			},
+		},
+	}
+}
+
+func resourceBootSourceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*client.Client)
+
+	bootsource, err := getBootSource(client)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(fmt.Sprintf("%v", bootsource.ID))
+
+	// We create by changing the url to the chosen value
+	bootsourceParams := entity.BootSourceParams{
+		URL: d.Get("url").(string),
```

Comment:
what about the keyring fields? we should try to set them if any of them is specified by the user